codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithydotnet/TypeConversionCodegen.java

```diff
@@ -251,6 +259,100 @@ public TypeConverter generateBooleanConverter(final BooleanShape booleanShape) {
         return buildConverterFromMethodBodies(booleanShape, fromDafnyBody, toDafnyBody);
     }
 
+    public String generateStringConstraints(final Shape shape, final String value) {
```

Comment:
Again, that's a different topic.
pull/257 is responsible for "we don't support that" errors.
This PR makes dotnet behave the same as Java.
I suppose that part pf pull/257 might be to rip these out, since the smithy file will have to remove any length constraints on strings.